python/core-sdk/alation_ai_agent_sdk/tools.py

```diff
@@ -312,3 +314,235 @@ def __init__(self, api: AlationAPI):
 
     def run(self, job_id: int) -> dict:
         return self.api.check_job_status(job_id)
+
+
+class GenerateDataProductTool:
+    def __init__(self, api: AlationAPI):
+        self.api = api
+        self.name = self._get_name()
+        self.description = self._get_description()
+        self._cached_schema = None  # Cache the schema to avoid repeated requests
+
+    @staticmethod
+    def _get_name() -> str:
+        return "generate_data_product"
```

Comment:
This doesn't actually generate data product, could it be confusing? Should we call it something like data_product_build_helper? Can't think of a shorter name.

Reply:
Nah, let's skip it. I can't think of a better name, and if Robb has reviewed it already I'm good.